### PR TITLE
CustomSingleChildLayout should be able to resize

### PR DESCRIPTION
**Pull request by abarth**
It's not correct to set sizedByParent for
RenderCustomSingleChildLayoutBox because the delegate's size function
might depend on information other than the incoming constraints.

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie @HansMuller

File: packages/flutter/lib/src/rendering/shifted_box.dart [hunk 1]
```diff
@@ -736,18 +736,54 @@ class RenderFractionallySizedOverflowBox extends RenderAligningShiftedBox {
...
+  /// Defaults to the given constraints.
   BoxConstraints getConstraintsForChild(BoxConstraints constraints) => constraints;
 
-  /// Returns the position where the child should be placed given the size of this object and the size of the child.
+  /// The position where the child should be placed.
+  ///
+  /// The `size` argument is the size of the parent, which might be different
+  /// from the value returned by [getSize] if that size doesn't satisfy the
+  /// constraints passed to [getSize]. The `childSize` argument is the size of
+  /// the child, which is required to satify the constrains returned by
```
Comment (Hixie) [inline packages/flutter/lib/src/rendering/shifted_box.dart]: s/is required to/will/

Comment (Hixie) [inline packages/flutter/lib/src/rendering/shifted_box.dart]: s/constrains/constraints/

**Comment by Hixie** [comment]
LGTM

**Pull request merged**